iodata/formats/gamess.py

```diff
@@ -0,0 +1,141 @@
+# IODATA is an input and output module for quantum chemistry.
+# Copyright (C) 2011-2019 The IODATA Development Team
+#
+# This file is part of IODATA.
+#
+# IODATA is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""GAMESS punch file format."""
+
+
+import numpy as np
+
+from ..docstrings import document_load_one
+from ..utils import angstrom, LineIterator
+
+
+__all__ = []
+
+
+PATTERNS = ['*.gamessout']
+
+
+def _read_data(lit: LineIterator) -> tuple:
+    """Extract ``title``, ``symmetry`` and ``symbols`` from the punch file."""
+    title = next(lit).strip()
+    symmetry = next(lit).split()[0]
+    if symmetry != "C1":
+        raise NotImplementedError("Only C1 symmetry is supported.")
```

Comment:
I'm not sure if we really only support C1 symmetry here. Do we know someone who can generate a few GAMESS punch files to check this?

Reply:
GAMESS supports a lot of symmetries (only Abelian I think though) but I'm not sure what we support. We may just allow a failure if it isn't C1? In HORTON more generally we don't support symmetries in any real sense, certainly not in `grid` or `gbasis` at this point. (I guess we can "copy over" symmetries.) My vague recollection (and a very very very tiny bit of googling seems to confirm this) is that this file contains only symmetry-unique atoms, so we would be incapable of supporting non-C1 symmetry without significant additional coding. So I feel like the "only C1 symmetry is supported" is reasonable for now. We could probably add a comment telling the user how to run a GAMESS calculation (or generate a PUNCH file) with no symmetry.

Reply:
@PaulWAyers Thanks for this. I was not aware of the symmetry-unique atoms. That would indeed make it hard at this stage to deal with anything else than C1.

Reply:
I jsut added a comment to the code, and updated the raised message.